### DISCUSSION TITLE
Best laptop for coding in 2024?

### DISCUSSION THREAD

**Discussion by 8-Pac** [community/community#131795] [Programming Help] [answered]
### Body

give me a suggestion for a laptop.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by LaluIman** [comment, marked as answer]
If you love apple invironment i suggest buying macbook pro, but if you not recognize or unfamilliar with it you can use window Dell XPS 15.

**Comment by Engineer-Hasibur-Rahman** [comment]
Apple MacBook Pro (M2 Pro/M2 Max):
Pros: Powerful performance, excellent battery life, stunning Retina display, macOS stability.
Cons: Expensive, limited port variety.

**Comment by ebndev** [comment]
Hi @8-Pac & @LaluIman, thanks for being a part of the GitHub Community!

While we appreciate you wanting to contribute to Discussions, please note: We made the decision to disable the ability to earn Achievements in our Community in order to discourage users from participating in coordinated or inauthentic activity like rapid questions and answers in order to earn badges. You can learn more about this decision in our announcement post here [Achievements will no longer be available in the Community](https://github.com/orgs/community/discussions/106536).

Note that GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this, and other related posts. 

Any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
